WcaOnRails/app/models/competition.rb

```diff
@@ -574,6 +588,10 @@ def events_with_rounds_with_results
       end
   end
 
+  def delegate_report
+    DelegateReport.find_by_competition_id(self.id) || DelegateReport.create!(competition_id: self.id, content: "Hey, enter some text here", posted_at: nil)
```

Comment:
Could this be a job for `find_or_create_by!`?

    DelegateReport. find_or_create_by!(self.id) do |dr|
      dr.competition_id = self.id
      dr.content = "Hey, enter some text here"
      dr.posted_at = nil
    end